Refactor JWT validation

silva-fj opened the pull request:
This pull request refactors the JWT authentication logic and improves the validation process. Now we validate that the tokens have the correct `typ` and remove the need to include the `sender` when verifying the tokens with `verify_auth_token_authentication`.

Note: This PR should be merged after https://github.com/litentry/heima/pull/3414 is merged. 

TODO: Update base branch after https://github.com/litentry/heima/pull/3414 is merged

silva-fj committed:
feat(jwt): require "sub" and "typ" claims in JWT validation

silva-fj committed:
update auth_token validation

silva-fj committed:
feat(auth): refactor JWT validation and usage in pumpx rpc methods

Refactor JWT validation in pumpx RPC methods to use the new
verify_auth_token_authentication function, which enforces the presence
of the "typ" claim and returns claims on success. Remove redundant key
handling and direct JWT decoding. Update submit_swap_order to check
user identity against the token subject.

kziemianek reviewed (commented):
We have many places where we set `NativeTask.sender` based on params provided to rpc method example (`PumpxAddWallet(Identity)` <- it holds only sender):

```
impl From<AddWalletParams> for NativeTaskWrapper<NativeTask> {
	fn from(p: AddWalletParams) -> Self {
		Self {
			task: NativeTask::PumpxAddWallet(Identity::from_web2_account(
				p.user_id.as_str(),
				Web2IdentityType::Pumpx,
			)),
			nonce: None,
			auth: Some(OmniAuth::AuthToken(p.auth_token)),
		}
	}
}
```

How do we make sure that `sub` from `JWT` and `sender` relates to same user ?
I don't know if it's right PR to address this but this one surerly removes check of `JWT` sub field.

silva-fj commented:
> We have many places where we set NativeTask.sender based on params provided to rpc method example (PumpxAddWallet(Identity) <- it holds only sender)

In this case we can just use the OmniAccount, in the handler we only use the sender to get it (`sender.to_omni_account()`), I'll check all the pumpx native tasks and update them.

> How do we make sure that `sub` from `JWT` and `sender` relates to same user ? I don't know if it's right PR to address this but this one surerly removes check of `JWT` sub field.

In the cases where we need the sender we can check `if claims.sub == sender.to_omni_account().to_hex()`

silva-fj commented:
Ok this will require more changes, I'll keep the `sub` validation, we can tackle this in another issue

silva-fj committed:
revert changes to pumpx rpc methods

silva-fj committed:
feat(auth): validate JWT subject matches user identity

kziemianek commented on tee-worker/omni-executor/rpc-server/src/verify_auth.rs:
```diff
@@ -100,9 +104,11 @@ pub fn verify_auth_token_authentication(
 	ctx: Arc<RpcContext>,
 	sender: &Identity,
```
What do you thing about getting rid of inner `sender.omni_account()` and require `omni_account: String` explicitly ?

silva-fj replied:
I've updated the code 👍🏼

silva-fj committed:
Merge branch 'p-1453-re-design-and-implement-the-msg-for-web3-authentication' into p-1479-check-the-typ-of-incoming-jwt-token

silva-fj committed:
refactor(auth): unify omni account hex usage in verification

Update authentication verification to consistently use the hex-encoded omni
account string instead of Identity references.

silva-fj committed:
Merge branch 'p-1453-re-design-and-implement-the-msg-for-web3-authentication' into p-1479-check-the-typ-of-incoming-jwt-token

silva-fj committed:
Merge remote-tracking branch 'origin/p-1453-re-design-and-implement-the-msg-for-web3-authentication' into p-1479-check-the-typ-of-incoming-jwt-token

silva-fj commented:
I will include al thesel changes in another PR
